sources/org.osbuild.ostree

```diff
@@ -114,14 +123,17 @@ class OSTreeSource(sources.SourceService):
     def fetch_one(self, checksum, desc):
         commit = checksum
         remote = desc["remote"]
+        subpath = remote.get("subpath") or []
+        subpath = [f'--subpath={p}' for p in subpath]
```

Comment:
Note that there is an upper limit to both the amount of arguments we can pass and the total size of these arguments. Especially with paths we could hit that (didn't do any napkin math) if used extensively.

I don't know if we want and should guard against this.

Reply:
It's definitely possible but I think we'd need dozens of paths to hit the limit (I also did not do any maths for this).  In reality, I think we're going to use this for 2-3 paths.

I don't know if we can guard against it in any meaningful way, though.  We don't want to set an arbitrary max limit on the number of paths, it would be too, well, arbitrary, so catching it at the schema level is probably not an option.  Anything at runtime only leaves us the option of failing the build, which is what will happen if the line is too long anyway.  But then, if we can catch it, a more relevant error message (as opposed to a stack trace with a slightly relevant error message somewhere at the top) would be a lot more helpful if it ever happens.

Reply:
I am not sure if I follow, any command will eventually fail on `ARG_MAX` kernel limit so it ends up with an error. Compared to a fixed limit that also ends with an error but slightly better error message. In my books, this is not worth the effort.

Reply:
There's an advantage to failing during schema validation before the build starts (its earlier, and we never call a program that'll fail).

However, it's a minor thing and if the usecase is not going to trigger that it's indeed: not worth the effort :)